google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/resources/RepositorySelector.java

```diff
@@ -91,13 +91,15 @@ public String getSelectedRepository() {
       return null;
     }
 
-    Enumeration repos = ((RepositoryPanel) panel).getRepositories().children();
-    while (repos.hasMoreElements()) {
-      TreeNode repo = (TreeNode) repos.nextElement();
-
-      if (repo instanceof RepositoryModelItem
-          && getText().equals(((RepositoryModelItem) repo).getRepositoryId())) {
-        return getText();
+    if (panel instanceof RepositoryPanel) {
```

Comment:
as discussed offline, panel gets instantiated differently if repo's are found vs if they are not found. So in the case of typing an invalid repo, the panel will be an instance of the empty panel type.